### PR TITLE
Enabling claims parameter on broker authentication requests.

### PR DESCRIPTION
**Pull request by trwalke**
Work in progress PR for claim support in android and iOS brokers.

Fix for https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1651

This will be accompanied by a manual E2E test

### CODE REVIEW

**Comment by trwalke** [comment]
@bgavrilMS Do you know how I can E2E test this?

File: src/client/Microsoft.Identity.Client/Internal/Broker/BrokerInteractiveRequestComponent.cs [hunk 1]
```diff
@@ -93,6 +92,7 @@ private async Task<MsalTokenResponse> SendAndVerifyResponseAsync()
             BrokerPayload.Add(BrokerParameter.Username, _authenticationRequestParameters.Account?.Username ?? string.Empty);
             BrokerPayload.Add(BrokerParameter.ExtraOidcScopes, BrokerParameter.OidcScopesValue);
             BrokerPayload.Add(BrokerParameter.Prompt, _interactiveParameters.Prompt.PromptValue);
+            BrokerPayload.Add(BrokerParameter.Claims, _authenticationRequestParameters.ClaimsAndClientCapabilities);
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/Internal/Broker/BrokerInteractiveRequestComponent.cs]: @bgavrilMS 
This looks like the reason iOS didnt work E2E with claims as the claims were simply never put into the request parameters for broker. Also, we may have to split up the client capabilities and claims on the AuthenticationRequestParameters.cs object into individual parameters depending on how the broker handles these.

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Internal/Broker/BrokerInteractiveRequestComponent.cs]: Yes, good point. Our MSAL differentiates between capabilities and claims already.

**Comment by bgavrilMS** [comment]
I don't know how to test claims. You could try to just request some claims (e.g. using the json from the spec). Perhaps the Android folks know more since they have added some more advanced features on this?

File: src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs [hunk 2]
```diff
@@ -429,7 +430,9 @@ private Bundle GetSilentBrokerBundle(IDictionary<string, string> brokerPayload)
                 ClientVersion = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.ClientVersion),
                 CorrelationId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.CorrelationId),
                 HomeAccountId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.HomeAccountId),
-                LocalAccountId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LocalAccountId)
+                LocalAccountId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LocalAccountId),
+                ForceRefresh = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.ForceRefresh),
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs:434]: Force refresh was missing here

File: src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs [hunk 3]
```diff
@@ -405,7 +405,8 @@ private Intent GetInteractiveBrokerIntent(IDictionary<string, string> brokerPayl
                 ClientAppVersion = Application.Context.PackageManager.GetPackageInfo(Application.Context.PackageName, PackageInfoFlags.MatchAll).VersionName,
                 ClientVersion = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.ClientVersion),
                 CorrelationId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.CorrelationId),
-                Prompt = "NONE"
+                Prompt = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.Prompt).ToUpperInvariant(),
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs:408]: Just found a bug in the broker where if you send the prompt without making the text uppercase the json parsing will sometimes fail on the broker side and crash the broker app. It just so happened that adding claims to the json trigged this bug. They are going to fix in the next release.

File: src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs [hunk 4]
```diff
@@ -429,7 +430,9 @@ private Bundle GetSilentBrokerBundle(IDictionary<string, string> brokerPayload)
                 ClientVersion = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.ClientVersion),
                 CorrelationId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.CorrelationId),
                 HomeAccountId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.HomeAccountId),
-                LocalAccountId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LocalAccountId)
+                LocalAccountId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LocalAccountId),
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs:433]: few additional  parameters that were missing here. these were also tested

**Comment by bgavrilMS** [comment]
Some of the unit tests are failing on this.

**Comment by bgavrilMS** [comment]
@trwalke - I incorporated these changes as part of the refactoring I've done related to broker request objects.

**Pull request closed without merging**